blockchain/cryptoProvider/cryptoComponentsHolder.go

```diff
@@ -0,0 +1,66 @@
+package cryptoProvider
+
+import (
+	crypto "github.com/ElrondNetwork/elrond-go-crypto"
+	"github.com/ElrondNetwork/elrond-go-crypto/signing"
+	"github.com/ElrondNetwork/elrond-go-crypto/signing/ed25519"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/core"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+)
+
+var (
+	suite  = ed25519.NewEd25519()
+	keyGen = signing.NewKeyGenerator(suite)
+)
+
+type cryptoComponentsHolder struct {
+	publicKey      crypto.PublicKey
+	privateKey     crypto.PrivateKey
+	addressHandler core.AddressHandler
+}
+
+// NewCryptoComponentsHolder returns a new cryptoComponentsHolder instance
+func NewCryptoComponentsHolder(skBytes []byte) (*cryptoComponentsHolder, error) {
+	privateKey, err := keyGen.PrivateKeyFromByteArray(skBytes)
+	if err != nil {
+		return nil, err
+	}
+	publicKey := privateKey.GeneratePublic()
+
+	publicKeyBytes, err := publicKey.ToByteArray()
+	if err != nil {
+		return nil, err
+	}
+	addressHandler := data.NewAddressFromBytes(publicKeyBytes)
+
+	return &cryptoComponentsHolder{
+		privateKey:     privateKey,
+		publicKey:      publicKey,
+		addressHandler: addressHandler,
+	}, nil
+}
+
+// GetPublicKey returns the held publicKey
+func (holder *cryptoComponentsHolder) GetPublicKey() crypto.PublicKey {
+	return holder.publicKey
+}
+
+// GetPrivateKey returns the held privateKey
+func (holder *cryptoComponentsHolder) GetPrivateKey() crypto.PrivateKey {
+	return holder.privateKey
+}
+
+// GetBech32 returns the held bech32 address
+func (holder *cryptoComponentsHolder) GetBech32() string {
+	return holder.addressHandler.AddressAsBech32String()
```

Comment:
this could have been cached in a string that was created on the constructor

Reply:
done